src/pages/components/file-upload.mdx

```diff
@@ -0,0 +1,96 @@
+---
+title: File upload
+description: >
+  File upload lets users select and upload files.
+package: '@zendeskgarden/react-forms'
+components:
+  - forms/src/elements/FileUpload.tsx
+---
+
+<Usage>
+  <Use>
+    <ul>
+      <li>To let users attach a file or image to something else</li>
+      <li>To let users add images to their avatars</li>
+    </ul>
+  </Use>
+</Usage>
+
+## How to use it
+
+### Default
+
+Users drag and drop files on to the target or click to select which files to upload.
+
+import Default from '../../examples/forms/file-upload/FileUpload.tsx';
+import DefaultCode from '!!raw-loader!../../examples/forms/file-upload/FileUpload.tsx';
+
+<CodeExample code={DefaultCode}>
+  <Default />
+</CodeExample>
+
+### Disabled
+
+A disabled File upload prevents user interaction. It doesn’t appear in the tab order and can’t receive focus.
+
+import Disabled from '../../examples/forms/file-upload/Disabled.tsx';
+import DisabledCode from '!!raw-loader!../../examples/forms/file-upload/Disabled.tsx';
+
+<CodeExample code={DisabledCode}>
+  <Disabled />
+</CodeExample>
+
+### Size
+
+File upload can be default or compact. Size affects spacing between the label/description and the touch target.
+
+import Size from '../../examples/forms/file-upload/Size.tsx';
+import SizeCode from '!!raw-loader!../../examples/forms/file-upload/Size.tsx';
+
+<CodeExample code={SizeCode}>
+  <Size />
+</CodeExample>
+
+## How to use it well
+
+### Put file requirements in hints
+
+<BestPractice>
+  <Do imageSource={props.data.fileUploadRequirements.childFile.childImageSharp.fluid} imageIsSquare>
+    <p>
+      <Markdown>Use `Hint` to describe image requirements and restrictions</Markdown>
+    </p>
+  </Do>
+</BestPractice>
+
+## Configuration
+
+<Configuration reactPackage={props.data.mdx.package} components={props.data.mdx.components} />
+
+The `FileUpload` component is a styled `div` which can be used with 3rd-party file upload libraries
+like [react-dropzone](https://github.com/react-dropzone/react-dropzone/).
+
+## API
+
+### FileUpload
+
+<PropSheet components={props.data.mdx.components} componentName="fileUpload" />
+
+import { graphql } from 'gatsby';
+
+export const pageQuery = graphql`
+  query($fileAbsolutePath: String) {
+    ...SidebarPageFragment
+    fileUploadRequirements: abstractAsset(
+      layerName: { eq: "components-forms-fileupload-file-requirements" }
```

Comment:
Not sure if this can be helped, as i think this is recurring issue with gatsby/abstract, but the image is very very small.

Reply:
This aligns with the image size shown in `Autocomplete` and other standalone `Do/Dont` areas. I haven't looked into this much, but I think the sizing is applied through the `BestPractices` component rather than Abstract.

Either way, this may be something we want to track separate.

Reply:
@steue the `max-height` for do/don't images was designed to be 160px. We may need to flex this design POV in order to accommodate other treatments. Please connect with @rossmoody, et al. to redefine and help us consider new or additional target sizes.